reactive/docker_registry.py

```diff
@@ -227,11 +225,25 @@ def remove_certs():
 
 
 @when('charm.docker-registry.configured')
-@when('leadership.is_leader')
 @when('website.available')
 def update_reverseproxy_config():
+    '''Configure a reverse proxy.
+
+    The oldest known registry peer will be configured as the primary proxied
+    service. Other peers will be configured as backup services which can take
+    over if the primary fails.
+    '''
     website = endpoint_from_flag('website.available')
+    peers = peer_units(peer_relation="peer")
     port = hookenv.config().get('registry-port')
+
+    # NB: use oldest peer charmhelper versus leadership to determine primary
+    # vs backup servers:
+    #  https://bugs.launchpad.net/layer-docker-registry/+bug/1815459
+    server_opts = ["check", "inter 2000", "rise 2", "fall 5", "maxconn 4096"]
+    if not oldest_peer(peers):
+        server_opts.append("backup")
```

Comment:
We need to make sure that all servers are in the same order on all haproxy units because should the non-backup server fail, haproxy will use the first server in the list with "backup".

Else we could use the "first" balance algo (https://cbonte.github.io/haproxy-dconv/1.6/configuration.html#4-balance) and assign an id to each server based on unit number or machine number for example.

Reply:
@axinojolais nice catch.  I'll fix this by reinstating the leadership guard for the website relation. Now that we know all the registry peers, we'll use the fact that there can only be 1 leader to ensure all haproxy units have the same servers configured.